server/handlerequest.go

```diff
@@ -34,41 +35,41 @@ func handleRequest(r *repo.Repo, handler Handler, jsonBytes []byte, source strin
 	// case HandlerGetRepo: // This case is handled prior to handleRequest being called.
 	// since the resulting bytes are written directly in to the http.ResponseWriter / net.Connection
 	case HandlerGetURIs:
-		getURIRequest := &requests.URIs{}
-		processIfJSONIsOk(json.Unmarshal(jsonBytes, &getURIRequest), func() {
+		getURIRequest := new(requests.URIs)
+		processIfJSONIsOk(json.NewDecoder(rdr).Decode(getURIRequest), func() {
```

Comment:
I would recommend to create the json decoder only once prior to the switch statement, 
and then call .Decode on it in each handler.

Reply:
Good spot! I've changed it.